tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,578 @@
+import { BooleanLike, classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { flow } from 'common/fp';
+import { map, filter, sortBy, sort } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Tabs, Stack, Box, Input, NoticeBox, Icon } from '../components';
+import { Window } from '../layouts';
+import { Food } from './PreferencesMenu/data';
+
+const TYPE_ICONS = {
+  'Can Make': 'utensils',
+  [Food.Alcohol]: 'wine-glass',
+  [Food.Breakfast]: 'sun',
+  [Food.Bugs]: 'bug',
+  [Food.Cloth]: 'tshirt',
+  [Food.Dairy]: 'cheese',
+  [Food.Fried]: 'fire',
+  [Food.Fruit]: 'apple-alt',
+  [Food.Gore]: 'skull',
+  [Food.Grain]: 'wheat-awn',
+  [Food.Gross]: 'trash',
+  [Food.Junkfood]: 'pizza-slice',
+  [Food.Meat]: 'bacon',
+  [Food.Nuts]: 'seedling',
+  [Food.Oranges]: 'apple-alt',
+  [Food.Pineapple]: 'apple-alt',
+  [Food.Raw]: 'drumstick-bite',
+  [Food.Seafood]: 'fish',
+  [Food.Sugar]: 'candy-cane',
+  [Food.Toxic]: 'biohazard',
+  [Food.Vegetables]: 'carrot',
+} as const;
+
+const CATEGORY_ICONS = {
+  'Can Make': 'utensils',
+  'Breads': 'bread-slice',
+  'Burgers': 'burger',
+  'Cakes': 'cake-candles',
+  'Egg-Based Food': 'egg',
+  'Frozen': 'ice-cream',
+  'Lizard Food': 'dragon',
+  'Meats': 'bacon',
+  'Mexican Food': 'pepper-hot',
+  'Misc. Food': 'bowl-food',
+  'Mothic Food': 'shirt',
+  'Pastries': 'cookie',
+  'Pies': 'chart-pie',
+  'Pizzas': 'pizza-slice',
+  'Salads': 'leaf',
+  'Sandwiches': 'hotdog',
+  'Seafood': 'fish',
+  'Soups': 'mug-saucer',
+  'Spaghettis': 'wheat-awn',
+} as const;
+
+type Ingredient = {
+  path: string;
+  name: string;
+  amount: number;
+  is_reagent: BooleanLike;
+};
+
+type Recipe = {
+  ref: string;
+  result: string;
+  name: string;
+  desc: string;
+  reqs: Ingredient[];
+  category: string;
+  foodtypes: string[];
+  non_craftable: BooleanLike;
+  is_reaction: BooleanLike;
+  steps: string[];
+  tool_paths: string[];
+  catalysts: Ingredient[];
+  machinery: string[];
+};
+
+type Diet = {
+  liked_food: string[];
+  disliked_food: string[];
+  toxic_food: string[];
+};
+
+type Data = {
+  // Dynamic
+  busy: BooleanLike;
+  display_compact: BooleanLike;
+  display_craftable_only: BooleanLike;
+  craftability: Record<string, BooleanLike>;
+  // Static
+  recipes: Recipe[];
+  categories: string[];
+  foodtypes: string[];
+  diet: Diet;
+};
+
+export const PersonalCooking = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
```

Comment:
please deconstruct your data here, same way as props.
should be able to delete all occurrences of `data.` thereafter

Reply:
I don't think it makes sense, since this data needs processing, and final variable name would clash with deconstructed names.

Reply:
Thanks! It's a lot cleaner now!

Reply:
> I don't think it makes sense, since this data needs processing, and final variable name would clash with deconstructed names.

I see what you mean! good point